Hatcher's acyclic type is acyclic (#1003)
UniMath/agda-unimath: src/synthetic-homotopy-theory/hatchers-acyclic-type.lagda.md

```agda
module _
  {l1 l2 : Level}
  ((A , a1 , a2 , r1 , r2) : algebra-Hatcher-Acyclic-Type l1)
  ((B , b1 , b2 , s1 , s2) : algebra-Hatcher-Acyclic-Type l2)
  where

  is-hom-pointed-map-algebra-Hatcher-Acyclic-Type : (A →∗ B) → UU l2
  is-hom-pointed-map-algebra-Hatcher-Acyclic-Type f =
    Σ ( map-Ω f a1 ＝ b1)
      ( λ u →
        Σ ( map-Ω f a2 ＝ b2)
          ( λ v →
            ( coherence-square-identifications
              ( ap (map-Ω f) r1)
              ( map-power-nat-Ω 5 f a1 ∙ ap (power-nat-Ω 5 B) u)
              ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
              ( s1)) ×
            ( coherence-square-identifications
              ( ap (map-Ω f) r2)
              ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
              ( ( map-power-nat-Ω 2 f (a1 ∙ a2)) ∙
                ( ap
                  ( power-nat-Ω 2 B)
                  ( ( preserves-mul-map-Ω f) ∙
                    ( horizontal-concat-Id² u v))))
              ( s2))))

  hom-algebra-Hatcher-Acyclic-Type : UU (l1 ⊔ l2)
  hom-algebra-Hatcher-Acyclic-Type =
    Σ ( A →∗ B) is-hom-pointed-map-algebra-Hatcher-Acyclic-Type
```

> ### The Hatcher acyclic type is the initial Hatcher acyclic algebra

Reviewer: Whoops, I just noticed that this header doesn't match the section's contents.
One possibility is
```suggestion
### The universal property of Hatcher's acyclic type

The universal property of Hatcher's acyclic type is that it is initial among
Hatcher acyclic algebras.
```
Sorry for the patchy review. I think it is ready to be merged after this one.

Reviewer: Actually, perhaps that's not a good header either, as the definition doesn't make reference to Hatcher's acyclic type

Reviewer: Here's another alternative, I'll let you decide.
```suggestion
### Initial Hatcher acyclic algebras

One characterization of Hatcher's acyclic type is through its universal property
as being the initial Hatcher acyclic algebra.
```

Author: Not at all patchy!

### Morphisms of types equipped with the structure of Hatcher's acyclic type

```agda
module _
  {l1 l2 : Level}
  ((A , a1 , a2 , r1 , r2) : algebra-Hatcher-Acyclic-Type l1)
  ((B , b1 , b2 , s1 , s2) : algebra-Hatcher-Acyclic-Type l2)
  where

>   Hom-algebra-Hatcher-Acyclic-Type-pointed-map : (A →∗ B) → UU l2

Reviewer: A better name for this thing is `is-hom-pointed-map-algebra-Hatcher-Acyclic-Type`. In particular, `hom` should not be capitalized.

Author: Ah, thanks for bringing this up. The issue I had with `is-hom-...` is that the type is not necessarily a proposition. Is it OK to use the prefix `is` in that case?

Reviewer: Yep. We for instance use `is-...` for `is-section` and `is-retraction`

Author: OK, thank you. Fixed now.

  is-hom-pointed-map-algebra-Hatcher-Acyclic-Type f =
    Σ ( map-Ω f a1 ＝ b1)
      ( λ u →
        Σ ( map-Ω f a2 ＝ b2)
          ( λ v →
            ( coherence-square-identifications
              ( ap (map-Ω f) r1)
              ( map-power-nat-Ω 5 f a1 ∙ ap (power-nat-Ω 5 B) u)
              ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
              ( s1)) ×
            ( coherence-square-identifications
              ( ap (map-Ω f) r2)
              ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
              ( ( map-power-nat-Ω 2 f (a1 ∙ a2)) ∙
                ( ap
                  ( power-nat-Ω 2 B)
                  ( ( preserves-mul-map-Ω f) ∙
                    ( horizontal-concat-Id² u v))))
              ( s2))))

  Hom-algebra-Hatcher-Acyclic-Type : UU (l1 ⊔ l2)

>   Hom-algebra-Hatcher-Acyclic-Type =

Reviewer: Sorry for being unclear, the same applies here: `hom` should be uncapitalized.

Author: Happy to use lowercase, but it thought it would be line with things like `Pointed-Type`? Can you explain the convention?

Reviewer: Well, it's complicated, and we are in a (slow) process of shaping out a more detailed naming convention. But basically, the capitalized part of a name should function as a namespace for the entry. In particular, it always comes last in the name, and there's at most one namespace for every entry.

If you want more context, Egbert wrote up a draft for a new naming convention a while ago that you can find here:
https://github.com/UniMath/agda-unimath/pull/793/files#diff-ae980e981cdc5f111930bf629a726443e1fb5cb0b251ab39539c20d8ff4598da

>     Σ ( map-Ω f a1 ＝ b1)
>               ( λ u →
>                 Σ ( map-Ω f a2 ＝ b2)
>                   ( λ v →
>                     ( coherence-square-identifications
>                       ( ap (map-Ω f) r1)
>                       ( map-power-nat-Ω 5 f a1 ∙ ap (power-nat-Ω 5 B) u)
>                       ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
>                       ( s1)) ×
>                     ( coherence-square-identifications
>                       ( ap (map-Ω f) r2)
>                       ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
>                       ( ( map-power-nat-Ω 2 f (a1 ∙ a2)) ∙
>                         ( ap
>                           ( power-nat-Ω 2 B)
>                           ( ( preserves-mul-map-Ω f) ∙
>                             ( horizontal-concat-Id² u v))))
>                       ( s2))))

Reviewer: ```suggestion
    Σ ( map-Ω f a1 ＝ b1)
      ( λ u →
        Σ ( map-Ω f a2 ＝ b2)
          ( λ v →
            ( coherence-square-identifications
              ( ap (map-Ω f) r1)
              ( map-power-nat-Ω 5 f a1 ∙ ap (power-nat-Ω 5 B) u)
              ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
              ( s1)) ×
            ( coherence-square-identifications
              ( ap (map-Ω f) r2)
              ( map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)
              ( ( map-power-nat-Ω 2 f (a1 ∙ a2)) ∙
                ( ap
                  ( power-nat-Ω 2 B)
                  ( ( preserves-mul-map-Ω f) ∙
                    ( horizontal-concat-Id² u v))))
              ( s2))))
```

Author: Ah, sorry copy-paste error I forgot to fix. Thanks!

Reviewer: No worries!

        ( is-torsorial-path refl)
```

### For a fixed pointed map, the `Hom-algebra-Hatcher-Acyclic-Type-pointed-map` family is [torsorial](foundation.torsorial-type-families.md)

In proving this, it is helpful to consider an equivalent formulation of
`Hom-algebra-Hatcher-Acyclic-Type-pointed-map` for which torsoriality is almost
immediate.

```agda
module _
  {l1 l2 : Level}
  ((A , a1 , a2 , r1 , r2) : algebra-Hatcher-Acyclic-Type l1)

>   ((B , b1 , b2 , s1 , s2) : algebra-Hatcher-Acyclic-Type l2)

Reviewer: This sort of pattern matching in a module declaration is unwound to `let`-expressions in Agda, which tend to be the opposite of helpful in interactive mode. Hence, it is generally inadvisable to use them. But since you've already finished the section I'm not necessarily asking you to change this. Perhaps @EgbertRijke has a stronger opinion on that though.

Author: I thought it might be OK because prior to my editing the file already used this.

Reviewer 2: Let's accept this for now and fix it later, but I agree with Fredrik that if this use of pattern matching messes up the way Agda displays goals then this problem should be addressed sometime. Thanks for noticing it Fredrik!

  where

  Hom-algebra-Hatcher-Acyclic-Type-pointed-map' : (A →∗ B) → UU l2
  Hom-algebra-Hatcher-Acyclic-Type-pointed-map' f =

>     Σ ( map-Ω f a1 ＝ b1)
>               ( λ u →
>                 Σ ( map-Ω f a2 ＝ b2)
>                   ( λ v →
>                     ( Id
>                       ( inv (map-power-nat-Ω 5 f a1 ∙ ap (power-nat-Ω 5 B) u) ∙
>                         ( ap (map-Ω f) r1 ∙
>                           (map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)))
>                       ( s1)) ×
>                     ( Id
>                       ( inv (map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v) ∙
>                         ( ap (map-Ω f) r2 ∙
>                           ( ( map-power-nat-Ω 2 f (a1 ∙ a2)) ∙
>                             ( ap
>                               ( power-nat-Ω 2 B)
>                               ( ( preserves-mul-map-Ω f) ∙
>                                 ( horizontal-concat-Id² u v))))))
>                       ( s2))))

Reviewer: ```suggestion
    Σ ( map-Ω f a1 ＝ b1)
      ( λ u →
        Σ ( map-Ω f a2 ＝ b2)
          ( λ v →
            ( Id
              ( inv (map-power-nat-Ω 5 f a1 ∙ ap (power-nat-Ω 5 B) u) ∙
                ( ap (map-Ω f) r1 ∙
                  (map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v)))
              ( s1)) ×
            ( Id
              ( inv (map-power-nat-Ω 3 f a2 ∙ ap (power-nat-Ω 3 B) v) ∙
                ( ap (map-Ω f) r2 ∙
                  ( ( map-power-nat-Ω 2 f (a1 ∙ a2)) ∙
                    ( ap
                      ( power-nat-Ω 2 B)
                      ( ( preserves-mul-map-Ω f) ∙
                        ( horizontal-concat-Id² u v))))))
              ( s2))))
```

        ( is-torsorial-path refl)
```

> ### For a fixed pointed map, the `is-hom-pointed-map-algebra-Hatcher-Acyclic-Type` family is [torsorial](foundation.torsorial-type-families.md)

Reviewer: Links in headers should probably be avoided. How about moving the link to where "torsoriality" is mentioned in the prose below
